db/change_cache.go

```diff
@@ -86,6 +87,7 @@ func (c *changeCache) Init(context *DatabaseContext, lastSequence SequenceID, on
 	c.onChange = onChange
 	c.channelCaches = make(map[string]*channelCache, 10)
 	c.receivedSeqs = make(map[uint64]struct{})
+	c.terminator = make(chan bool, 1) // give it buffer size of 1 so something trying to send won't block forever
```

Comment:
We don't ever actually want to send to this channel, though, right?  Seems like giving it a buffer of 1 might result in future errors slipping through.

Reply:
Good catch.  Originally I was sending to the channel, then later signalled via closing the channel.  Will fix.